File: Aci.Flex.Plugins.AnsellAfe.Web/CodeGen/SaveAfeApiController.ts
module Flex.ApiClient.SaveAfeApi {

export function SaveAfe(afe: Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData): Q.Promise<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>
{
	var baseUrl = $.url("portal");
	var typeHint = "Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails";
	var targetUrl = baseUrl + "api/saveafeapi/SaveAfe";
    var payload = ko.toJSON(afe);
	return Aci.Flex.Portal.ServiceHelper.PostData<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>(targetUrl, payload);
}

export function ResolveSaveAfe(afe: Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData, resolveTo: KnockoutObservable<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>) : Q.Promise<boolean>
{
	  return SaveAfe(afe).then	((e) => 
	  {
		resolveTo(e); 
		return true;
	}, (f) => {return false;});
}

export function SaveDraftAfe(afe: Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData): Q.Promise<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>
{
	var baseUrl = $.url("portal");
	var typeHint = "Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails";
	var targetUrl = baseUrl + "api/saveafeapi/SaveDraftAfe";
    var payload = ko.toJSON(afe);
	return Aci.Flex.Portal.ServiceHelper.PostData<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>(targetUrl, payload);
}

export function ResolveSaveDraftAfe(afe: Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData, resolveTo: KnockoutObservable<Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails>) : Q.Promise<boolean>
{
	  return SaveDraftAfe(afe).then	((e) => 
	  {
		resolveTo(e); 
		return true;
	}, (f) => {return false;});
}


}

File: Aci.Flex.Plugins.AnsellAfe.Web/Controllers/Api/SaveAfeApiController.cs
using Aci.Flex.Core.Logging;
using Aci.Flex.Interfaces;
using Aci.Flex.Plugins.AnsellAfe.Web.Models;
using Aci.Flex.Portal.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;
using System.Web.Http;
using System.Web.Http.Description;

namespace Aci.Flex.Plugins.AnsellAfe.Web.Controllers.Api
{
    [ExportAsFlexPlugin(typeof(SaveAfeApiController), typeof(IFlexApiControllerV2), false)]
    [RoutePrefix("api/saveafeapi")]
    public class SaveAfeApiController : BaseAnsellAfeApiController, IFlexApiControllerV2
    {
        [Route("saveafe")]
        [ResponseType(typeof(Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails))]
        [HttpPost]
        public async Task<IHttpActionResult> SaveAfe(Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData afe)
        {
            try
            {
                var returnDetails = await AnsellAfeServiceHelper.UseAnsellAfeServiceAsync(s => s.SaveAfeAsync(m_UserIdentifier, afe, false));
                return Ok(returnDetails);
            }
            catch (Exception ex)
            {
                Log.Exception(ex);
                return Ok(new Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails() { Success = false });
            }
        }
        [Route("savedraftafe")]
        [ResponseType(typeof(Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails))]
        [HttpPost]
        public async Task<IHttpActionResult> SaveDraftAfe(Aci.Flex.Plugins.AnsellAfe.Server.Objects.AfeSubmitData afe)
        {
            try
            {
                var returnDetails = await AnsellAfeServiceHelper.UseAnsellAfeServiceAsync(s => s.SaveAfeAsync(m_UserIdentifier, afe, true));
                return Ok(returnDetails);
            }
            catch (Exception ex)
            {
                Log.Exception(ex);
                return Ok(new Aci.Flex.Plugins.AnsellAfe.Server.Objects.SaveAfeReturnDetails() { Success = false });
            }
        }
    }
}